### PR TITLE
Improve TypeInfo test to also test the model type info construction

### PR DESCRIPTION
**Pull request by vietj**
A set of TypeInfo improvements, mostly missing tests and the TypeInfo based on java.lang.reflect was missing some handling.

### CODE REVIEW

**Comment by vietj** [comment]
please someone review @purplefox @cescoffier @pmlopes , thanks

File: src/main/java/io/vertx/codegen/TypeInfo.java [hunk 1]
```diff
@@ -66,13 +67,25 @@ public static TypeInfo create(Type type) {
...
         } else {
           ClassKind kind = Helper.getKind(classType::getAnnotation, fqcn);
+          List<TypeParamInfo.Class> typeParams = new ArrayList<>();
+          int index = 0;
+          for (java.lang.reflect.TypeVariable<? extends java.lang.Class<?>> var : classType.getTypeParameters()) {
+            typeParams.add(new TypeParamInfo.Class(classType.getName(), index++, var.getName(), Collections.<Variance>emptySet()));
+          }
           if (kind == ClassKind.API) {
-            return new Class.Api(fqcn, true, Collections.emptyList(), null, null, null, module, false);
+            java.lang.reflect.TypeVariable<java.lang.Class<ReadStream>> classTypeVariable = ReadStream.class.getTypeParameters()[0];
```
Comment (cescoffier) [inline src/main/java/io/vertx/codegen/TypeInfo.java]: Why using the FQN here ?

File: src/test/java/io/vertx/test/codegen/testtype/EnumHolder.java [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+import java.util.concurrent.TimeUnit;
+
+/**
+ * @author <a href="mailto:julien@julienviet.com">Julien Viet</a>
+ */
+public interface EnumHolder {
+
+
+  ValidEnum apiEnum();
+  TimeUnit otherEnum();
```
Comment (cescoffier) [inline src/test/java/io/vertx/test/codegen/testtype/EnumHolder.java]: What about list and set of enum ? Should it be tested too ?

**Comment by cescoffier** [comment]
2 very small comments. Everything else is fine.

File: src/main/java/io/vertx/codegen/TypeInfo.java (hunk 1 above)
Comment (vietj) [inline src/main/java/io/vertx/codegen/TypeInfo.java]: why wouldn't we use the FQN ?

File: src/test/java/io/vertx/test/codegen/testtype/EnumHolder.java (hunk 2 above)
Comment (vietj) [inline src/test/java/io/vertx/test/codegen/testtype/EnumHolder.java]: right, they shall be tested rather in CollectionHolder, I will add test (I added enum test after Collection tests)

File: src/main/java/io/vertx/codegen/TypeInfo.java (hunk 1 above)
Comment (cescoffier) [inline src/main/java/io/vertx/codegen/TypeInfo.java]: Is it necessary for ` java.lang.reflect.TypeVariable` ?

Comment (vietj) [inline src/main/java/io/vertx/codegen/TypeInfo.java]: there is lot of code to read, so it's not clear what FQN refers to (in clear : I'm lost :-) ) can you give more context ?

Comment (cescoffier) [inline src/main/java/io/vertx/codegen/TypeInfo.java]: It's really a detail. 

In `java.lang.reflect.TypeVariable<java.lang.Class<ReadStream>> classTypeVariable`, why not using `TypeVariable<Class<ReadStream>> ...`. Is there a conflict ?

Comment (vietj) [inline src/main/java/io/vertx/codegen/TypeInfo.java]: ah yes , there is one : with `TypeInfo.Class`

**Pull request merged**